libimage/layer_tree.go

```diff
@@ -148,7 +148,7 @@ func (r *Runtime) layerTree(ctx context.Context, images []*Image) (*layerTree, e
 func (t *layerTree) layersOf(image *Image) []*storage.Layer {
 	var layers []*storage.Layer
 	node := t.node(image.TopLayer())
-	for node != nil {
+	for node != nil && node.layer != nil {
```

Comment:
I am not familiar with the code but this does not seem to do the same
I would assume the condition to be inside the loop:
```
if  node.layer != nil {layers = append(layers, node.layer)}
```

Because with your change the loop would exit as soon as one layer is nil not necessarily loop over all parents. I don't know if this makes a difference in practise but I think it would be safer to check this inside the loop.

Reply:
Thanks, done :+1: